### DISCUSSION TITLE
TurbineLite Bug and reinstalling FOQUS bug

### DISCUSSION THREAD

**Discussion by gauridalwankar1617** [CCSI-Toolset/FOQUS#1249] [Q&A] [answered]
I am currently trying to install FOQUS but have encountered persistent issues preventing the software from opening. I have attached a file that outlines the specific error I'm facing, which I have not been able to resolve.

Initially, I encountered issues with TurbineLite. After installing it, I was unable to locate the relevant folder or GUI associated with its installation. While attempting to troubleshoot, FOQUS stopped working unexpectedly, and since then, I’ve continued to experience the same error, even after several reinstallations from scratch.

I did attempt to install FOQUS on another laptop, where it worked fine; however, the TurbineLite problem persists across both machines. Could you please assist me in debugging this issue as soon as possible? I need the software for some urgent work and would really appreciate any help here!

I have also attached the log file from FOQUS for your reference.

[FOQUS_Error.txt](https://github.com/user-attachments/files/17497617/FOQUS_Error.txt)
[TurbineLite_Error.txt](https://github.com/user-attachments/files/17497618/TurbineLite_Error.txt)

![image](https://github.com/user-attachments/assets/6055edd2-cbe3-401a-abdb-55e5ac974328)

**Comment by bpaul4** [comment]
Hello @gauridalwankar1617, can you please share some further details regarding your Anaconda environment? Activate your environment, run `pip list`, and find your installed package versions for `ccsi-foqus` and `TurbineClient` as well as your Python and operating system versions. Please also check your installed program list for your installed versions of SimSinter and TurbineLite.

**Reply by bpaul4** [reply]
As the error appears to be an authentication error, can you please check if the Turbine Client is running? Go to your Task Manager, Services, find "TurbineWebAPIService", right click and select "Restart".

**Reply by gauridalwankar1617** [reply]
I checked the versions of CCSI-FOQUS (3.22.0) and Turbine Client (4.0.3) and restarted the TurbineWebAPIService. But FOQUS still won’t open. It was working fine earlier but suddenly stopped, and I can’t figure out why. I’ve attached the error message I’m getting when trying to launch FOQUS:
[(ccsi-foqus) Error.txt](https://github.com/user-attachments/files/17526245/ccsi-foqus.Error.txt)

I tried uninstalling and reinstalling everything, but the same issue keeps happening. Any ideas on how to fix this?

**Comment by boverhof** [comment]
The turbine configuration log line should be a file ( eg. `turbine.cfg` ), but in your debug log it is a directory.  That is definitely screwing things up, maybe you can open up the help menu and configure it?  Maybe create a new work space on startup?  
## Your Output
```
2024-10-23 13:11:16,731 - DEBUG - foqus.foqus_lib.framework.sim.turbineConfiguration - turbine configuration="C:\Program Files (x86)\Turbine\Lite"
```

## My setup
```
...
2024-10-14 10:59:07,423 - DEBUG - foqus.foqus_lib.framework.plugins.pluginSearch - check plugin available: pymodel_test
2024-10-14 10:59:07,423 - DEBUG - foqus.foqus_lib.framework.plugins.pluginSearch - check plugin available: steam_cycle
2024-10-14 10:59:07,423 - DEBUG - foqus.foqus_lib.framework.sim.turbineConfiguration - turbine configuration="turbine.cfg"
2024-10-14 10:59:07,423 - DEBUG - foqus.foqus_lib.framework.sim.turbineConfiguration - turbine configuration application url="http://localhost:8000/TurbineLite/application/"
2024-10-14 10:59:07,423 - DEBUG - foqus.foqus_lib.framework.sim.turbineConfiguration - turbine configuration url="http://localhost:8000/TurbineLite"
2024-10-14 10:59:07,439 - DEBUG - foqus.foqus_lib.foqus - Load Flowsheet Session: None
2024-10-14 10:59:07,439 - DEBUG - foqus.foqus_lib.foqus - LoadValues Flowsheet: None
2024-10-14 10:59:07,439 - DEBUG - foqus.foqus_lib.foqus - Run Option: None
2024-10-14 10:59:07,439 - DEBUG - foqus.foqus_lib.foqus - Starting GUI
```

**Comment by boverhof** [comment, marked as answer]
If all else fails you should be able to edit the `.foqus.cfg` file directly
I'm guessing your turbConfig below is the problem.

```
C:\Users\boverhof>echo %APPDATA%\.foqus.cfg
C:\Users\boverhof\AppData\Roaming\.foqus.cfg

C:\Users\boverhof>more %APPDATA%\.foqus.cfg
{
  "working_dir": "C:\\Users\\boverhof\\Desktop\\foqus_test1",
  "simsinter_path": "C:\\Program Files\\CCSI\\SimSinter",
  "psuade_path": "C:\\Users\\boverhof\\.conda\\envs\\foqus\\bin\\psuade.exe",
  "foqusLogLevel": 10,
  "turbLogLevel": 30,
  "foqusLogToConsole": true,
  "turbLogToConsole": true,
  "foqusLogToFile": true,
  "turbLogToFile": true,
  "foqusLogFile": "foqus.log",
  "turbineLogFile": "turbine.log",
  "logFormat": "%(asctime)s - %(levelname)s - %(name)s - %(message)s",
  "alamo_path": "",
  "aspenVersion": 2,
  "turbLiteHome": "C:\\Program Files\\Turbine\\Lite",
  "logRotate": false,
  "maxLogSize": 5.0,
  "compactSession": true,
  "backupSession": false,
  "rScriptPath": "C:\\Program Files\\R\\R-3.1.2\\bin\\x64\\Rscript.exe",
  "recentlyOpenedFiles": [
    "C:\\Users\\boverhof\\Desktop\\FOQUS\\examples\\test_files\\OUU_Remote_Turbine_BFB\\BFB_OUU_06_09_2020.foqus",
    "C:\\Users\\boverhof\\Desktop\\FOQUS\\examples\\test_files\\UQ\\Rosenbrock.foqus",
    "C:\\Users\\boverhof\\Desktop\\FOQUS\\examples\\tutorial_files\\Flowsheets\\Tutorial_4\\Simple_flow.foqus",
    "C:\\Users\\boverhof\\Desktop\\FOQUS\\examples\\other_files\\ML_AI_Plugin\\Supported_Keras_Models\\mea_column_model.foqus"
  ],
  "turbConfig": "turbine.cfg",
  "turbConfigCluster": "turbine_aws.cfg",
  "runFlowsheetMethod": 0,
  "settingsInWDir": false,
  "numRecentFiles": 5,
  "turbineRemoteCheckFreq": 10.0,
  "turbineRemoteReSub": 0
}

C:\Users\boverhof>
```

**Reply by gauridalwankar1617** [reply]
Thank you so much!
This resolved the issue.
